core-bundle/src/Resources/contao/drivers/DC_Folder.php

```diff
@@ -1964,7 +1972,7 @@ public function source()
 			throw new InternalServerErrorException('File "' . $this->intId . '" does not exist.');
 		}
 
-		$this->denyAccessUnlessGranted(ContaoCorePermissions::DC_ACTION_EDIT, new DataContainerSubject($this->strTable, $this->intId));
+		$this->denyAccessUnlessGranted(ContaoCorePermissions::DC_PREFIX . $this->strTable, new UpdateAction($this->strTable, array('id' => $this->intId)));
```

Comment:
does `DC_Folder` have a `$this->strTable` at all? What if the file is excluded from DBAFS? Or should the action data source be the parent folder in this case (though I sure would keep the attribute with table name)?

Reply:
It does. All the fields are coming from `tl_files`.